lib/ddtrace/contrib/que/tracer.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'ddtrace/contrib/analytics'
+
+module Datadog
+  module Contrib
+    module Que
+      # Tracer is a Que's server-side middleware which traces executed jobs
+      class Tracer
+        def call(job)
+          request_span = generate_spans(job)
+          set_sample_rate(request_span)
+          Contrib::Analytics.set_measured(request_span)
+
+          yield
+        rescue StandardError => e
+          request_span.set_error(e) unless request_span.nil?
+          raise e
+        ensure
+          request_span.finish if request_span
```

Comment:
I noticed you are using the explicit pattern of: starting the trace, handling error, and finishing it explicitly.
This is a more brittle pattern than using the `tracer.trace(...) { |span| yield }` block, as the block ensures the spans is always finished and handles error propagation for you.

I don't see a strong reason to use the explicit operations, instead of the `tracer.trace{}` block syntax here, so I'd recommend the change the block pattern.

**But**, if you have a reason to keep this explicit pattern, please let me know and we can work with it no problem.

Reply:
No particular reason, I went off the example I have seen in the `sneaker` tracer you suggested.

I myself find explicit style much more readable and maintainable since all the steps are available right in front of the developer to read and understand (less magic and tracing). But I would be more than happy to switch the pattern to comply with your repo style.

Thank you for pointing that out.

Reply:
updated now : )